DAQ/src/ArtBinaryPacketsFromDigis_module.cc

```diff
@@ -0,0 +1,1113 @@
+//
+// EDProducer module for converting straw hit digis into DTC formatted
+// packets (which are stored in DataBlock data products)
+//
+//
+// Original author Tomonari Miyashita
+//
+//
+
+// C++ includes.
+#include <iostream>
+#include <string>
+#include <cmath>
+
+#include <math.h>
+
+
+// Framework includes.
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "art/Framework/Principal/Provenance.h"
+//geometry
+#include "GeometryService/inc/GeomHandle.hh"
+#include "GeometryService/inc/DetectorSystem.hh"
+#include "CalorimeterGeom/inc/DiskCalorimeter.hh"
+
+// mu2e-artdaq-core includes
+#include "mu2e-artdaq-core/Overlays/ArtFragment.hh"
+#include "mu2e-artdaq-core/Overlays/ArtFragmentReader.hh"
+
+//pci_linux_kernel_module includes
+#include "dtcInterfaceLib/DTC.h"
+
+// Mu2e includes.
+#include "RecoDataProducts/inc/StrawHitCollection.hh"
+#include "RecoDataProducts/inc/StrawDigiCollection.hh"
+#include "RecoDataProducts/inc/CaloDigiCollection.hh"
+//#include "DAQDataProducts/inc/DataBlockCollection.hh"
+
+#include "SeedService/inc/SeedService.hh"
+
+#include <fstream>
+#include <stdexcept>
+
+
+// Definitions and typedefs needed for compatibility with HLS codeblock
+#define NUM_PRESAMPLES 4
+#define START_SAMPLES 5 // 0 indexed
+#define NUM_SAMPLES 15
+#define LOWER_TDC 16000
+#define UPPER_TDC 64000
```

Comment:
Two things:

Always prefer to use constexpr over #define.   Remember that #define is NOT part of C++ it is a C-preprocessor directive.   Therefore many of the safety features of the compiler are not available.

As targets of opportunity, please change this everywhere in your code.

In any case these values should not be hard coded.  Other code will want to know them as well.  So  they belong somewhere that many modules have access to them.   Unless you can make a very compelling case that these are fixed for all time, the only place for them is in Proditions.